cf-agent/cf-agent.c

```diff
@@ -796,8 +798,12 @@ static void ThisAgentInit(void)
 */
 
     snprintf(filename, CF_BUFSIZE, "%s/cfagent.%s.log", GetLogDir(), VSYSNAME.nodename);
+    ToLowerStrInplace(filename);
     MapName(filename);
 
+    mode_t current_umask;
+    umask(current_umask = umask(0777));
```

Comment:
Add either an explaining comment, or a function `nonatomic_getumask()`.

Reply:
If you don't create a function, I would prefer something like this:
```C
    const mode_t current_umask = umask(0777); // Gets and changes umask
    umask(current_umask);  // Restores umask
```

Reply:
done